substrate/primitives/core/src/bandersnatch.rs

```diff
@@ -394,277 +333,199 @@ pub mod vrf {
 	}
 
 	impl VrfPublic for Public {
-		fn vrf_verify(&self, data: &Self::VrfSignData, signature: &Self::VrfSignature) -> bool {
-			const _: () = assert!(MAX_VRF_IOS == 3, "`MAX_VRF_IOS` expected to be 3");
-			let pre_outputs_len = signature.pre_outputs.len();
-			if pre_outputs_len != data.inputs.len() {
+		fn vrf_verify(&self, data: &VrfSignData, signature: &VrfSignature) -> bool {
+			use ark_vrf::ietf::Verifier;
+			let Ok(public) =
+				bandersnatch::Public::deserialize_compressed_unchecked(self.as_slice())
+			else {
 				return false
-			}
-			// Workaround to overcome backend signature generic over the number of IOs.
-			match pre_outputs_len {
-				0 => self.vrf_verify_gen::<0>(data, signature),
-				1 => self.vrf_verify_gen::<1>(data, signature),
-				2 => self.vrf_verify_gen::<2>(data, signature),
-				3 => self.vrf_verify_gen::<3>(data, signature),
-				_ => unreachable!(),
-			}
+			};
+			let Ok(proof) =
+				ark_vrf::ietf::Proof::deserialize_compressed_unchecked(signature.proof.as_slice())
+			else {
+				return false
+			};
+			public
```

Comment:
I think prove's logic here should be as simple as verify logic and complication moved to library? why prove can't be similar to verify?

Reply:
shouldn't ths be called `vrf_verify` instead of `verify` . I know it is about the library but still.

Reply:
`ark-vrf` has only VRF prove+verify. So `verify` is just the verification of a vrf proof